web/shapeworks/src/components/Analysis/AnalysisTab.vue

```diff
@@ -76,87 +71,117 @@ export default {
             openTab,
             currentlyCaching,
             pca,
-            groups
+            groups,
+            resetForm,
+            jobAlreadyDone
         }
     },
 }
 </script>
 
 <template>
-    <div v-if="taskData" class="messages-box pa-3">
-        Running analysis...
-        <div v-if="taskData.error">{{ taskData.error }}</div>
-        <v-progress-linear v-else :value="taskData.percent_complete"/>
-        <br />
-    </div>
-    <div class="pa-3" v-else-if="analysis">
-        Review shape analysis
-        <v-tooltip bottom>
-            <template v-slot:activator="{ on, attrs }">
-                <v-icon
-                    v-bind="attrs"
-                    v-on="on"
-                >
-                mdi-information
-                </v-icon>
-            </template>
-            <span>
-               <v-subheader>Last analysis run at {{ analysis.modified }}</v-subheader>
-            </span>
-        </v-tooltip>
-        <v-expansion-panels v-model="openTab">
+    <div v-if="jobAlreadyDone('optimize')">
+        <v-expansion-panels :value="analysis ? 1 : 0">
             <v-expansion-panel>
-                <v-expansion-panel-header>
-                    View PCA
-                </v-expansion-panel-header>
-                <v-expansion-panel-content>
-                    <PCA ref="pca" :currentTab="currentTab" :openTab="openTab"/>
-                    <charts :charts="analysis.charts"/>
-                </v-expansion-panel-content>
-            </v-expansion-panel>
-            <v-expansion-panel :disabled="analysis.groups.length <= 0" id="groups-panel">
-                <v-expansion-panel-header>
-                    Group Difference
-                </v-expansion-panel-header>
+                <v-expansion-panel-header>Controls</v-expansion-panel-header>
                 <v-expansion-panel-content>
-                    <groups ref="groups" :currentTab="currentTab" :openTab="openTab"/>
+                    <task-info
+                        taskName="analyze"
+                        :formData="formData"
+                        @resetForm="resetForm"
+                        :disabled="!analysis"
+                    />
+                    <v-text-field
+                        v-model.number="formData.range"
+                        type="number"
+                        step="1.0"
+                        max="10.0"
+                        min="1.0"
+                        label="Std. Dev. Range"
+                        hide-details
+                        :disabled="!analysis"
+                    />
+                    <v-text-field
+                        v-model.number="formData.steps"
+                        type="number"
+                        step="1.0"
+                        min="1.0"
+                        max="100.0"
+                        label="Number of steps"
+                        hide-details
+                        :disabled="!analysis"
+                    />
```

Comment:
Isn't this missing step size?

Reply:
The step size can be intuited from the range and number of steps, and it appeared that step size was not sent to the analyze task as an argument. Previously, line 152 of `PCA.vue` sent the task request like this: `await spawnJob("analyze", {"range": range, "steps": numSteps})`. 
The previous form only had fields for `numSteps` and `range`, and the `numSteps` field included this change function: `@change="params.step = (params.range) / params.numSteps"`. I assume the `range` field should've had this clause too, but the resulting value wasn't used for the task anyway. It looks like the analyze task does this division on its own.

Reply:
Oh, that's right. I added that division to ShapeWorks because it was a pain to do it in the client side. My bad, changes are fine!